data/frameProcessor/src/ExcaliburProcessPlugin.cpp

```diff
@@ -311,14 +308,54 @@ namespace FrameProcessor
         throw std::runtime_error(msg.str());
       }
 
-      // Allocate buffer to receive reordered image.
-      reordered_image = (void*)malloc(output_image_size);
-      if (reordered_image == NULL)
+
+      // Create and populate metadata for the reordered frame
+      FrameMetaData frame_meta;
```

Comment:
I've just spotted the default constructor for FrameMetaData sets the following:

            data_type_(raw_8bit),
            compression_type_(no_compression),

I think this should be changed to:

            data_type_(raw_unknown),
            compression_type_(unknown_compression),

so nothing is assumed prior to the explicit set calls.

Reply:
I'm not setting the compression type in this plugin either so will have to do that explicitly. I think this touches on the discussion we had a few weeks back about constructor arguments vs a set of required method calls after construction, although I'm not sure it changes the conclusion. Will you open a PR on odin-data for the change and I'll update this one?

Reply:
> I'm not setting the compression type in this plugin either so will have to do that explicitly. I think this touches on the discussion we had a few weeks back about constructor arguments vs a set of required method calls after construction, although I'm not sure it changes the conclusion. Will you open a PR on odin-data for the change and I'll update this one?

Sorry - ignore that, there's already a constructor for FrameMetaData that takes a list of arguments, I was confusing it with the discussion we had over constructing Frame objects. I'll modify this PR to either use the explicit constructor or ensure I set compression type.

Reply:
But I probably do still want to open a PR on odin-data to modify the constructor which doesn't take the list of arguments to set compression and type to unknown, rather than a random assumption ...

Reply:
Yeah that's sensible. I meant ignore my ramblings, you were right all along 👍